In [1]:
import pandas as pd

1. Find out how many countries are there in the dataset.

In [5]:
country_details = pd.read_excel("Dataset_w6.xlsx")
country_details

country             Region  \
0                           Afghanistan       SouthernAsia   
1                               Albania     SouthernEurope   
2                               Algeria     NorthernAfrica   
3                                Angola       MiddleAfrica   
4                   Antigua and Barbuda          Caribbean   
..                                  ...                ...   
191  Venezuela (Bolivarian Republic of)       SouthAmerica   
192                            Viet Nam  South-easternAsia   
193                               Yemen        WesternAsia   
194                              Zambia      EasternAfrica   
195                            Zimbabwe      EasternAfrica   

     Surface area (km2)  Population in thousands (2017)  \
0                652864                           35530   
1                 28748                            2930   
2               2381741                           41318   
3               1246700                           29784   
4                   442                             102   
..                  ...                             ...   
191              912050                           31977   
192              330967                           95541   
193              527968                           28250   
194              752612                           17094   
195              390757                           16530   

     GDP per capita (current US$)  Agricultural production index   \
0                           623.2                             125   
1                          3984.2                             134   
2                          4154.1                             160   
3                          4714.1                             175   
4                         14764.5                              88   
..                            ...                             ...   
191                       11068.9                             118   
192                        2067.9                             136   
193                        1106.4                             136   
194                        1311.1                             179   
195                         890.4                              99   

     International trade: Imports (million US$)  
0                                          3568  
1                                          4669  
2                                         47091  
3                                          8790  
4                                           491  
..                                          ...  
191                                       16324  
192                                      174111  
193                                        6861  
194                                        7442  
195                                        5212  

[196 rows x 7 columns]

In [6]:
country_details['country'].count()

196

   196 countries are present in the dataset.

2.  Find out how many countries with respect to the region are available in the dataset

In [8]:
country_details.groupby('Region')['country'].count()

Region
Caribbean            16
CentralAmerica        8
CentralAsia           5
EasternAfrica        17
EasternAsia           7
EasternEurope        10
Melanesia             5
Micronesia            4
MiddleAfrica          9
NorthernAfrica        5
NorthernAmerica       4
NorthernEurope       10
Oceania               2
Polynesia             5
South-easternAsia    11
SouthAmerica         12
SouthernAfrica        5
SouthernAsia          9
SouthernEurope       12
WesternAfrica        15
WesternAsia          18
WesternEurope         7
Name: country, dtype: int64

3. On average which region has the highest 
        a. GDP per capita (current US$)

In [10]:
sorted_by_gdp=country_details[['Region','GDP per capita (current US$)']].groupby('Region').mean().sort_values('GDP per capita (current US$)',ascending=False)
sorted_by_gdp

GDP per capita (current US$)
Region                                         
NorthernAmerica                    57659.000000
WesternEurope                      55387.128571
Oceania                            44823.250000
NorthernEurope                     42098.220000
EasternAsia                        27967.614286
WesternAsia                        17013.938889
Caribbean                          15135.562500
SouthernEurope                     14462.741667
South-easternAsia                  10299.027273
EasternEurope                       9292.490000
Melanesia                           9261.960000
Polynesia                           8686.520000
SouthAmerica                        8499.775000
CentralAmerica                      6348.137500
Micronesia                          5008.300000
CentralAsia                         4329.880000
SouthernAfrica                      4191.900000
NorthernAfrica                      3934.960000
MiddleAfrica                        3928.833333
SouthernAsia                        2968.400000
EasternAfrica                       2088.852941
WesternAfrica                       1013.140000

NorthernAmerica has the highest GDP per capita (current US$) : 57659.000000

    b. International trade: Imports (million US$)

In [20]:
sorted_by_international_trade=country_details[['Region','International trade: Imports (million US$)']].groupby('Region').mean().sort_values('International trade: Imports (million US$)',ascending=False)
sorted_by_international_trade

International trade: Imports (million US$)
Region                                                       
NorthernAmerica                                 663552.250000
EasternAsia                                     451995.000000
WesternEurope                                   407498.428571
NorthernEurope                                  113407.200000
Oceania                                         112914.500000
South-easternAsia                                98324.636364
EasternEurope                                    85243.900000
SouthernEurope                                   76666.666667
SouthernAsia                                     58199.333333
CentralAmerica                                   57031.500000
WesternAsia                                      46025.555556
NorthernAfrica                                   35034.400000
SouthAmerica                                     32885.666667
SouthernAfrica                                   18005.600000
CentralAsia                                      12063.400000
WesternAfrica                                     4898.333333
EasternAfrica                                     4726.176471
MiddleAfrica                                      4381.222222
Caribbean                                         3057.312500
Melanesia                                         2896.400000
Polynesia                                          432.000000
Micronesia                                          99.000000

NorthernAmerica has the highest International trade: Imports (million US$) : 663552.250000

4. Find out which region has more consistent with respect to GDP per capita (current US$)

In [16]:
country_details.groupby('Region')['GDP per capita (current US$)'].std().sort_values().head(1)

Region
WesternAfrica    815.095399
Name: GDP per capita (current US$), dtype: float64